sw/airborne/modules/guidance_opticflow/guidance_opticflow_hover.c

```diff
@@ -149,28 +149,42 @@ void guidance_h_module_run(bool in_flight)
  * Update the controls on a new VELOCITY_ESTIMATE ABI message.
  */
 static void stabilization_opticflow_vel_cb(uint8_t sender_id __attribute__((unused)),
-    uint32_t stamp, float vel_x, float vel_y, float vel_z, float noise)
+    uint32_t stamp, float vel_x, float vel_y, float vel_z, float noise_x, float noise_y, float noise_z)
 {
   /* Check if we are in the correct AP_MODE before setting commands */
   if (autopilot_get_mode() != AP_MODE_MODULE) {
     return;
   }
 
-  /* Calculate the error */
-  float err_vx = opticflow_stab.desired_vx - vel_x;
-  float err_vy = opticflow_stab.desired_vy - vel_y;
+  if (noise_x >= 0.f)
+  {
+    /* Calculate the error */
+    float err_vx = opticflow_stab.desired_vx - vel_x;
 
-  /* Calculate the integrated errors (TODO: bound??) */
-  opticflow_stab.err_vx_int += err_vx / 512;
-  opticflow_stab.err_vy_int += err_vy / 512;
+    /* Calculate the integrated errors (TODO: bound??) */
+    opticflow_stab.err_vx_int += err_vx / 512;
```

Comment:
what is this 512 ? If it is the related to the general periodic freq, it shouldn't be hard-coded. Or at least configurable.

Reply:
I guess it probably is but I didn't change this, only added a tab.

IMHO this entire module doesn't make much sense. This functionality should be possible with just normal control loops. I also see that this module isn't included in any airframe. I am in favour of deleting this altogether.

Reply:
I don't mind deleting old code. I guess it was a demo module made by Christophe or Freek. I let you guys decide if it is worth keeping it.